lib/pipefy_message/consumer.rb

```diff
@@ -80,12 +80,23 @@ def process_message
           logger.info(log_context({
                                     log_text: "Message received by poller to be processed by consumer",
                                     received_message: payload,
-                                    metadata: metadata
+                                    received_metadata: metadata
                                   }, context, correlation_id, event_id))
 
           retry_count = metadata["ApproximateReceiveCount"].to_i - 1
-          obj.perform(payload,
-                      { retry_count: retry_count, context: context, correlation_id: correlation_id })
+
+          begin
+            obj.perform(payload,
+                        { retry_count: retry_count, context: context, correlation_id: correlation_id })
+          rescue StandardError => e
+            logger.error(log_context({
+                                       received_message: payload,
+                                       received_metadata: metadata,
+                                       log_text: "Consumer #{obj.name}.perform method failed to process "\
+                                                 "received_message with #{e.inspect}"
+                                     }, context, correlation_id, event_id))
+            raise e
```

Comment:
**question:** @jleber (but also everyone), one thing I was thinking about during our daily when we talked about the retry mechanism is what happens should `perform` fail. It looks like this won't trigger the retry mechanism, as we've already `yield`ed from the broker `poller` method, which is the one that `throw`s stuff to be retried, in a broker-specific way. Should we retry in this case? Have we already reached a decision on how to deal with processing (not polling) failures?

Reply:
Maybe I didn't get your point, but I tested in my local environment (forced a `raise StandardError` in the `consumer.perform` and it retries after 30s).

We still have to think, desing and implement error scenarios yet...

Reply:
I created a card https://app.pipefy.com/open-cards/566652748 to further discussion on this issue. Just for the sake of moving forward with the improvements of this PR.